packs/data/equipment-effects.db/effect-trinity-geode-armor-greater.json

```
@@ -0,0 +1,71 @@
+{
+    "_id": "haywlcUtG6hV1LAy",
+    "img": "systems/pf2e/icons/equipment/other/spellhearts/trinity-geode.webp",
+    "name": "Effect: Trinity Geode - Armor (Greater)",
+    "system": {
+        "badge": null,
+        "description": {
+            "value": "<p>Granted by @UUID[Compendium.pf2e.equipment-srd.Trinity Geode (Greater)]{Trinity Geode (Greater)}</p>\n<p>After you cast an earth spell by Activating the <em>geode</em>, you gain resistance 3 to physical damage (except adamantine) until the end of your next turn, or double the resistance for a non-cantrip spell.</p>"
+        },
+        "duration": {
+            "expiry": "turn-end",
+            "sustained": false,
+            "unit": "rounds",
+            "value": 1
+        },
+        "level": {
+            "value": 8
+        },
+        "rules": [
+            {
+                "choices": [
+                    {
+                        "label": "PF2E.SpecificRule.TrinityGeode.Cantrip",
+                        "value": "cantrip"
+                    },
+                    {
+                        "label": "PF2E.SpecificRule.TrinityGeode.Non-Cantrip",
+                        "value": "non-cantrip"
+                    }
+                ],
+                "key": "ChoiceSet",
+                "label": "PF2E.SpecificRule.TrinityGeode.Label",
+                "prompt": "PF2E.SpecificRule.TrinityGeode.Prompt",
+                "rollOption": "effectTrinityGeodeArmor"
+            },
+            {
+                "except": "adamantine",
+                "key": "Resistance",
+                "type": "physical",
+                "value": 3
+            },
+            {
+                "except": "adamantine",
+                "key": "Resistance",
+                "predicate": [
+                    "effectTrinityGeodeArmor:non-cantrip"
+                ],
+                "type": "physical",
+                "value": 6
+            }
```

```suggestion
            {
                "choices": [
                    {
                        "label": "PF2E.SpecificRule.TrinityGeode.Cantrip",
                        "value": 3
                    },
                    {
                        "label": "PF2E.SpecificRule.TrinityGeode.Non-Cantrip",
                        "value": 6
                    }
                ],
                "key": "ChoiceSet",
                "label": "PF2E.SpecificRule.TrinityGeode.Label",
                "prompt": "PF2E.SpecificRule.TrinityGeode.Prompt",
                "flag": "spellType"
            },
            {
                "except": "adamantine",
                "key": "Resistance",
                "type": "physical",
                "value": "{item|flags.pf2e.ruleSelections.spellType}"
            }
```
We can make it a bit shorter by just selecting the value directly, but also roll options should always be in `kebab-case`. This is because as we parse roll options we don't want to try and parse other case styles. If roll options have a predictable case then they are easier to do more complicated things with in the code.

---

Good suggestion!